### ISSUE TITLE
NVDA reading PowerPoint slide titles twice

### ISSUE THREAD

**Issue by tvenie** [nvaccess/nvda#9379]

**Comment by tvenie** [comment]
Is NVDA supposed to read each title of a PowerPoint slide twice? By default, it is reading the slide titles twice at the beginning of each slide. I changed the order of the objects in the selection pane, but that only changed when it read the title for the second time. Is there a way to edit my slides so that NVDA will read the slide titles only once?

**Comment by Brian1Gaff** [comment]
I think like a number of people you are having issues with the template. I 
tend to export the template to a text editor using cut and paste, edit it 
then when going to the new ticket  edelete the template and paste the one I 
did off line. Just saying.


Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by Adriani90** [comment]
I can reproduce this issue as follows:
1. Open Powerpoint
2. on the first slide press tab and type "hello world"
3. Press escape to exit the edit mode
4. Press f5 to start presentation

Actual behavior: NVDA reports "hello world hello world"
Expected behavior: NVDA should only report "hello world"

cc: @lukaszgo1, @Qchristensen

**Comment by DrSooom** [comment]
@Adriani90: Does NVDA speak "Slide 1, Hello World" if you replace the title text field with a normal text field? I guess that this behavior you encounter is the expected one, because NVDA reads firstly the name of the slide and then the content of the first element/text field. Well, if the text field is a title text field, its content is identical to the name of the slide.

But please correct me, if I'm wrong, because I'm using Microsoft PowerPoint since 2010 extremely seldom.

**Comment by Adriani90** [comment]
Thanks Daniel for the suggestion. INdeed this is only repeated by NVDA if the slide is a title slide with title fields. Thus, powerpoint already defines that if the slide is a title slide, then its name will be the text which is written in the first title field. Thus, NVDA reads both slide title and the text in the title field.
@Qchristensen it would be very interesting to know how is this displayed visually? Are there two labels for a slide displayed if the slide has a title field?

**Comment by DrSooom** [comment]
There are three areas in Microsoft PowerPoint 2003, 2007 and 2010:
1. On the left side of the screen an overview with all slides and their title. If I'm not totally wrong it's possible here to toggle between list view and thumbnail view.
2. On the largest area, which is located from the middle to the right, you can see and edit the slide.
3. And directly below this large area there is a small area on the bottom of the screen for slide notes. If you're using two monitors, then the slides in presentation mode are shown on the one and their notes on the other screen. The slide title itself isn't visible in presentation mode when it is presented in full screen – but on the other screen with the slide notes.

@Qchristensen: Please correct me if I'm wrong. The last time I created a presentation (for school) was in 2009 – as a sighted person.

**Comment by Qchristensen** [comment]
TL;DR: The original issue is correct and the best solution is probably just for us not to read the title of each slide if we're going to go on and read the contents when running the slide show.

@DrSooom that is all correct.

@Adriani90 Visually a title placeholder doesn't look any different to any other placeholder with text in it (Title placeholder text isn't automatically bulleted like content placeholder text is, though it is possible to turn bullets on or off in any text placeholder).

The only place you can visually see anything different about titles, is if you view the outline view (alt+v, p, o, because of course it is).  In that view, the title of each slide is bold.

Interestingly (and getting further off the track), if you add a new text box to a slide (alt+n, x), the text you type in that box does not appear in the outline, even though other title and content text does appear.

Back to the issue, and when presenting a slideshow, NVDA reads the title of the slide, then the whole contents of the slide from back to front (objects can be aligned in front or behind each other - eg you might have a photograph and put text over the top of it - in that case you'd want the text "in front of" the photograph).  Normally everything gets read in a fairly logical order, but, if you selected the title placeholder and "bring to front", it would actually get read after the contents even without moving it.  (That also affects the tab order when navigating the contents in edit mode).

As a practical solution to this issue though, I would suggest if NVDA is going to read the contents of the slide, then don't read the title - since the title is almost always going to be the first thing read after that.

**Comment by DrSooom** [comment]
Thus I suggest a checkbox for this – as there are reasons not to locate the title text field at the lowest level. I created a grouped rectangle which contained four different forms, to animate them all together on just two slides. And this grouped rectangle was located behind the title text field, which had another animation. Yes, especially around 2005 I did some very fancy stuff with PowerPoint. And I also reached the borders of PowerPoint 2003. And before somebody ask: The level order of the elements on a slide is extremely important.

**Comment by Adriani90** [comment]
I propose to not report the slide title at all in presentation mode, but only the slide number and the content of the slide when switching slides in presentation mode.
 It seems Powerpoint assumes the title placeholder is the title of the slide and is exposed together with the slide number like "slide 1 hello" and then NVDA reports the actual content of the placeholders on the slide which starts with the title.
The reporting of slide number and title makes sense in all other views, but not in presentation mode.
@michaelDCurran is it possible to adjust what NVDA gets from the accessibility API when using presentation mode? I gues it gets slide number, slide  title and the whole content of the slide when changing the slide.
